cmd/all-in-one/main.go

```diff
@@ -362,3 +365,13 @@ func initTracer(metricsFactory metrics.Factory, logger *zap.Logger) io.Closer {
 	opentracing.SetGlobalTracer(tracer)
 	return closer
 }
+
+// Utility function to decide listening address based on port (deprecated flags) or host:port (new flags)
+func getAddressFromCLIOptions(port int, addr string, deprecatedFlag string, logger *zap.Logger) string {
```

Comment:
Note that the old flags are still defined with default values `flags.Int(collectorPort, ports.CollectorTChannel,...)`, so the value will not be 0 here.

I think you can change the default values for deprecated flags to 0's, since the new host-port flags will have the same default.

Reply:
PS - this should've come up in the unit tests, could you check why it didn't?